### PR TITLE
ci: adjust the ci workflows to current stable versions

### PR DESCRIPTION
**Pull request by TomerFi**
- ci: update release workflow runner, actions, and go versions
- ci: update acceptance workflow actions, python, node, and go versions
- ci: update build workflow actions, linting, and go versions
- chore: fixed lint errors reporeted golangci-lint
- ci: golint is deprecated and archived, vet and staticcheck are the recomended replacement

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -7,15 +7,15 @@ on:
...
+    runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up Go
-      uses: actions/setup-go@v2
+      uses: actions/setup-go@v4
       with:
-        go-version: 1.16
+        go-version: "1.20"
```
Comment (xieshenzh) [inline .github/workflows/release.yml:16]: Should we use 1.18 instead?

**Review by xieshenzh — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (TomerFi) [reply .github/workflows/release.yml:16]: Thank you @xieshenzh .
I think we can keep it 1.20, this sets the version we build with, this shouldn't affect the target version.

We currently targeting 1.15, if we want, we can bump it to 1.18 [here](https://github.com/fabric8-analytics/cli-tools/blob/main/go.mod#L3), but probably it'll be better to do it in a different PR.

Comment (TomerFi) [reply .github/workflows/release.yml:16]: We can discuss the target version offline, so we can be sure that's what we want to do.
Regardless, the version we build with, shouldn't have any affect.

Comment (TomerFi) [reply .github/workflows/release.yml:16]: @xieshenzh 
When I think about it, you made a good point. Especially post ADR-0006...

While it's ok to build and test everything with 1.20.
The `gomanifest`  tool should be verified with 1.15, but I don't see we have any test or workflow handling this.

I'm gonna put this PR on hold until I'll dd a workflows running 1.15 and verifying `gomanifest`.

Comment (TomerFi) [reply .github/workflows/release.yml:16]: @xieshenzh 
I added a workflow for building the `gomanifest` tool on multiple operating systems using go 1.15.
image

> The Build GoManifest job is running 1.15.

**Review by xieshenzh — APPROVED** [review]

**Pull request merged**